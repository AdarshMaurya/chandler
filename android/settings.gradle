rootProject.name = 'Monica'

include ':app'
