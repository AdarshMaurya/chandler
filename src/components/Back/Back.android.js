import React from 'react';
import PropTypes from 'prop-types';
import {TouchableWithoutFeedback} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {styles} from './Back.styles';

export const Back = ({onPress}) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <Icon name="close" size={38} color="black" style={styles.icon} />
  </TouchableWithoutFeedback>
);

Back.propTypes = {
  onPress: PropTypes.func.isRequired,
};
