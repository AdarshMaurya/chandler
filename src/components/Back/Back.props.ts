export interface IBackProps {
  onPress: (...args) => any;
}
