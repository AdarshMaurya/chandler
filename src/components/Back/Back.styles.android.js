import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  icon: {
    position: 'absolute',
    top: 8,
    right: 8,
  },
});
