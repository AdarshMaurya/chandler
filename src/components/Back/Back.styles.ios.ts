import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  icon: {
    position: 'absolute',
    top: 25,
    left: 15,
  },
});
