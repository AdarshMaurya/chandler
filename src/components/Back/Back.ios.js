import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import {styles} from './Back.styles';
import {hitSlop} from 'theme';

export const Back = ({onPress}) => (
  <TouchableOpacity onPress={onPress} style={styles.icon} hitSlop={hitSlop}>
    <Icon name="ios-arrow-back" size={35} color="black" />
  </TouchableOpacity>
);

Back.propTypes = {
  onPress: PropTypes.func.isRequired,
};
