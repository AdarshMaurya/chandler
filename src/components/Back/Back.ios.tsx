import React, {PureComponent} from 'react';
import {TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import {hitSlop} from '@theme';
import {IBackProps} from './Back.props';
import {styles} from './Back.styles.ios';

export class Back extends PureComponent<IBackProps> {
  public render() {
    const {onPress} = this.props;
    return (
      <TouchableOpacity onPress={onPress} style={styles.icon} hitSlop={hitSlop}>
        <Icon name="ios-arrow-back" size={35} color="black" />
      </TouchableOpacity>
    );
  }
}
