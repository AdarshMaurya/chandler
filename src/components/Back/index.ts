import {Platform} from 'react-native';
import {Back as BackAndroid} from './Back.android';
import {Back as BackIOS} from './Back.ios';

export const Back = Platform.OS === 'ios' ? BackIOS : BackAndroid;
