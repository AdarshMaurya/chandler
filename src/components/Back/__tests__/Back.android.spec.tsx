import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Back} from '../Back.android';

describe('Components', () => {
  describe('Back', () => {
    describe('Android', () => {
      it('should renders correctly', () => {
        const onPress = jest.fn();
        expect(toJson(shallow(<Back onPress={onPress} />))).toMatchSnapshot();
      });

      it('should call onPress when taping the icon', () => {
        const onPress = jest.fn();
        const button = shallow(<Back onPress={onPress} />);
        button.prop('onPress')();
        expect(onPress).toHaveBeenCalledTimes(1);
        expect(onPress).toHaveBeenCalledWith();
      });
    });
  });
});
