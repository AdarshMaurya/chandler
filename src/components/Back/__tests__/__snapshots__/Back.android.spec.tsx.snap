// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Back Android should renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    color="black"
    name="close"
    size={38}
    style={
      Object {
        "position": "absolute",
        "right": 8,
        "top": 8,
      }
    }
  />
</TouchableWithoutFeedback>
`;
